### ISSUE TITLE
Pod spec dependencies do not work with pod versions that have letters in them

### ISSUE THREAD

**Issue by mergesort** [CocoaPods/CocoaPods#5859]
* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

At least I suspect it's related to having letters. Either way, the `5.0.0-beta.3` is causing the problem. See below.

## What did you do?

Created a podspec for a local CocoaPod that reads like this.

```
Pod::Spec.new do |spec|
  spec.name         = 'redacted'
  spec.summary      = 'redacted'
  spec.version      = '1.0'
  spec.license      = { :type => 'MIT' }
  spec.authors      =  { 'redacted' => 'redacted' }
  spec.source_files = '*.swift'
  spec.ios.deployment_target = '9.0'
  spec.requires_arc = true
  spec.framework    = 'Foundation'
  spec.homepage = 'redacted'
  spec.source  = { :git => 'redacted }

  spec.dependency 'BrightFutures', '5.0.0-beta.3'
end
```

I also tried `v5.0.0-beta3`, because that's the official version listed on the [Bright Futures repo](https://github.com/Thomvis/BrightFutures/releases).

Run `pod install`

## What did you expect to happen?

Install the Bright Futures pod dependency correctly.

## What happened instead?

```
Fetching podspec for `API` from `./Teachable/Development Pods/API`
[!] Unable to satisfy the following requirements:

- `BrightFutures (= 5.0.0-beta.3)` required by `API (1.0)`

None of your spec sources contain a spec satisfying the dependency: `BrightFutures (= 5.0.0-beta.3)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.

Note: as of CocoaPods 1.0, `pod repo update` does not happen on `pod install` by default.
```

### Stack

```
   CocoaPods : 1.1.0.beta.2
        Ruby : ruby 2.1.3p242 (2014-09-19 revision 47630) [x86_64-darwin14.0]
    RubyGems : 2.2.2
        Host : Mac OS X 10.12 (16A319)
       Xcode : 8.0 (8A218a)
         Git : git version 2.2.1
Ruby lib dir : /Users/mergesort/.rvm/rubies/ruby-2.1.3/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 93a6b23278be743f69e8118f4d74fc3ac5a41fe2
               mergesort - https://github.com/mergesort/Communicado/ @ a0be6707160935436a00d01c16a219b6b90efbe4
```

### Installation Source

```
Executable Path: /Users/mergesort/.rvm/gems/ruby-2.1.3/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.1.0
```

**Comment by mergesort** [comment]
Tldr; I'm an idiot. The version was not pushed to a public repo yet, which explains why it worked in my podfile but not in the podspec.
